docs/development-process.md

```diff
@@ -0,0 +1,48 @@
+## Firefly Development process
+
+- _Note on examples_
+    - This example shows the `dev` branch, if you are working on a rc branch always use the rc branch instead of `dev`
+    - This example development branch will be `firefly-123456-table-sort`
+- Branch naming scheme 
+    - Branch Scheme: `Ticket Name` - `word or two description`
+    - Such As: Ticket fix as a table feature might be: `Firefly-123456-table-sort`
+
+### Process
+1. Create a branch using the naming scheme above from the `dev` branch
+    - Create the branch from the base branch
+        - goto base branch: `git checkout dev`
+        - create and change to the new branch: `git checkout -b firefly-123456-table-sort`
+    - Do the work and commit often.
+        - _develop code_
+        - `git commit -a`
+2. Rebased you to `dev`
+    - This will get any new work done by other developers
+        - `git checkout dev`
+        - `git pull origin dev`
+        - `git checkout firefly-123456-table-sort`
+        - `git rebase dev`
+    - Test your code again
+4. Create a pull request on github - when work is done
+    - Push the branch - `git push  origin firefly-123456-table-sort`
+    - Go to github and create a pull request with that branch.
+    - Build a test using Jenkins and put the URL in you PR.
+    - Get feedback, do updates, commits, push the changes.
+        - `git commit -a`
+        - `git push  origin firefly-123456-table-sort`
+5. When Pull Request is approved...
+    - Switch to `dev` and pull
+        - `git checkout dev`
+        - `git pull origin dev`
+        - `git checkout firefly-123456-table-sort`
+    - Rebase
+        - `git rebase -i dev`
+        - While rebasing squash the commit, the `git rebase -i` will put you in and editor to select the commit to squash
+    - Force push your rebased branch
+        - `git push -f origin firefly-123456-table-sort`
+    - goto github
+    - do the basic merge
+    - modify message to start with check ex- `Firefly-123456: merge pull request...`
```

Comment:
`start with check ex-`?  Do you mean to say, `start with ticket name follow by comments, i.e. Firefly-123456: merge...`

Reply:
yes, thanks